src/core/server/context/container/context.ts

```diff
@@ -178,21 +171,21 @@ export interface IContextContainer<THandler extends HandlerFunction<any>> {
 }
 
 /** @internal */
-export class ContextContainer<THandler extends HandlerFunction<any>>
+export class ContextContainer<THandler extends RequestHandler>
   implements IContextContainer<THandler> {
   /**
    * Used to map contexts to their providers and associated plugin. In registration order which is tightly coupled to
    * plugin load order.
    */
   private readonly contextProviders = new Map<
-    keyof HandlerContextType<THandler>,
+    string,
     {
-      provider: IContextProvider<THandler, keyof HandlerContextType<THandler>>;
+      provider: IContextProvider<any, any>;
```

Comment:
Super-NIT: I think this generalization of the types makes TS lose the references to `THandler`. And that is causing the _arity_ problems down the line. IMO, we can revisit these types later to unblock the TS References migrations.